src/extension/src/TelemetryWriter.py

```diff
@@ -66,9 +66,13 @@ def __ensure_message_restriction_compliance(self, full_message):
     def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informational, task_name=Constants.TELEMETRY_TASK_NAME):
         """ Creates and writes event to event file after validating none of the telemetry size restrictions are breached """
         try:
-            if self.events_folder_path is None or not os.path.exists(self.events_folder_path) or not Constants.TELEMETRY_ENABLED_AT_EXTENSION:
+
+            if self.events_folder_path is None or not Constants.TELEMETRY_ENABLED_AT_EXTENSION:
                 return
 
+            if not os.path.exists(self.events_folder_path):
+                os.mkdir(self.events_folder_path)
+
```

Comment:
Please sync with @rane-rajasi for a better place to put this. The write-event function is called a bunch of times during execution and this (folder existence and creation if necessary) should only be evaluated once during the start of each execution.

Reply:
Had a discussion over call, this should be moved to setup_telemetry() in ActionHandler.py. Please also test this on a VM

Reply:
Moved to `setup_telemetry()` in ActionHandler.py.